### PR TITLE
#3841 - Hotfix: Name change for CRA Response file

### PR DESCRIPTION
**Pull request by bidyashish**
Acceptance Criteria
- [x] Update CRA Response Integration job to search for new file name 
    - [x] [CRA Environment Code]BCSA#####.TXT
- [x] No change on request - confirmed with IMB CRA income verification request file name can remain the same. 

**Context:** 
DEV: ABCSA#####.TXT
TEST: ABCSA#####.TXT
PROD: PBCSA#####.TXT

**Current File Names for CRA Income:** 
- CCRA_REQUEST_${this.craConfig.environmentCode}${sequenceFile}.DAT
- /CCRA_RESPONSE_[\w]*\.txt/i

*note
Regex test

```
const environmentCode = 'A';
const regex = new RegExp(`[${environmentCode}]BCSA\\d{5}\\.TXT`, 'i');
const testString = 'ABCSA00001.TXT';

console.log(regex.test(testString)); // true
```

### CODE REVIEW

File: sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts [hunk 1]
```diff
@@ -198,7 +198,7 @@ export class CRAIncomeVerificationProcessingService {
   async processResponses(): Promise<ProcessSftpResponseResult[]> {
     const remoteFilePaths = await this.craService.getResponseFilesFullPath(
       this.ftpResponseFolder,
-      /CCRA_RESPONSE_[\w]*\.txt/i,
+      /[AP]BCSA\d{5}\.TXT/i,
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts:201]: `[AP]` should be replaced by `config.craIntegration.environmentCode`.

File: sources/packages/backend/libs/integrations/src/cra-integration/cra.integration.service.ts [hunk 2]
```diff
@@ -113,8 +119,13 @@ export class CRAIntegrationService extends SFTPIntegrationBase<CRASFTPResponseFi
     fileName: string;
     filePath: string;
   } {
-    const sequenceFile = sequence.toString().padStart(5, "0");
-    const fileName = `CCRA_REQUEST_${this.craConfig.environmentCode}${sequenceFile}.DAT`;
+    const fileNameBuilder = new StringBuilder();
+    fileNameBuilder.append(this.esdcConfig.environmentCode);
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/cra-integration/cra.integration.service.ts:123]: Is it not supposed to be `this.craConfig.environmentCode`?

File: sources/packages/backend/libs/integrations/src/cra-integration/cra.integration.service.ts
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/cra-integration/cra.integration.service.ts:1]: Is this file expected to be changed?
The AC states: 
"No change on request - confirmed with IMB CRA income verification request file name can remain the same."

**Review by andrewsignori-aot — COMMENTED** [review]
Please take a look at the comments, if I am missing something, please let me know.

File: sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts [hunk 3]
```diff
@@ -198,7 +198,10 @@ export class CRAIncomeVerificationProcessingService {
   async processResponses(): Promise<ProcessSftpResponseResult[]> {
     const remoteFilePaths = await this.craService.getResponseFilesFullPath(
       this.ftpResponseFolder,
-      /CCRA_RESPONSE_[\w]*\.txt/i,
+      new RegExp(
+        `[${this.configService.craIntegration.environmentCode}]BCSA\\d{5}\\.TXT`,
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts:202]: No need for square brackets. Please adjust as below.
```ts
`^${this.esdcConfig.environmentCode}EDU.PBC.DIS.D[\w]*\.[0-9]*`
```

Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts:202]: BCSA here does not represent the CRA area code from `this.configService.craIntegration.programAreaCode`?

Comment (andrewsignori-aot) [reply sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts]: There is no confirmation at this moment that programAreaCode should be used, hence moving ahead with the hardcoded value for the hotfix seems to be ok.

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍

File: sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts (hunk 3 above)
Comment (dheepak-aot) [reply sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts]: Can we remove the remove `CRA_PROGRAM_AREA_CODE` from the release instruction then?

File: sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts [hunk 4]
```diff
@@ -198,7 +198,10 @@ export class CRAIncomeVerificationProcessingService {
   async processResponses(): Promise<ProcessSftpResponseResult[]> {
     const remoteFilePaths = await this.craService.getResponseFilesFullPath(
       this.ftpResponseFolder,
-      /CCRA_RESPONSE_[\w]*\.txt/i,
+      new RegExp(
+        `${this.configService.craIntegration.environmentCode}BCSA\\d{5}\\.TXT`,
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/cra-integration/cra-income-verification.processing.service.ts:202]: 👍

**Review by dheepak-aot — APPROVED** [review]
Looks good 👍

**Pull request merged**